core/src/test/java/fj/control/db/TestDbState.java

```diff
@@ -0,0 +1,53 @@
+package fj.control.db;
+
+import fj.Unit;
+import fj.data.Option;
+import org.junit.Test;
+
+import java.sql.*;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+
+public class TestDbState {
+    @Test
+    public void testWriter() throws SQLException {
+        int TEN = 10;
+        DbState writer = DbState.writer(DbState.driverManager("jdbc:h2:mem:"));
+
+        DB<Unit> setup = new DB<Unit>() {
+            @Override
+            public Unit run(Connection c) throws SQLException {
```

Comment:
Thank you for the pointer. I'm running into issues with propagating the SQLException, so I would propose to commit as-is, and discuss in a followup PR.